src/App.tsx

```diff
@@ -14,37 +16,105 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const REVERSE = false;
+
+enum SortType {
+  RESET_VALUE = '',
+  SORT_BY_NAME = 'name',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface SortCriteria {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function sortGoods(
+  goods: string[],
+  { sortBy: criteria, isReversed }: SortCriteria,
+) {
+  let sortedGoods;
+
+  switch (criteria) {
+    case SortType.SORT_BY_NAME:
+      sortedGoods = goods.toSorted((a, b) => a.localeCompare(b));
+      break;
+    case SortType.SORT_BY_LENGTH:
+      sortedGoods = goods.toSorted((a, b) => a.length - b.length);
+      break;
+
+    default:
+      sortedGoods = [...goods];
+  }
+
+  if (isReversed) {
+    sortedGoods = sortedGoods.toReversed();
+  }
+
+  return sortedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.RESET_VALUE);
+  const [isReversed, setIsReversed] = useState(REVERSE);
+
+  const sortedGoods = sortGoods(goodsFromServer, { sortBy, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': SortType.SORT_BY_NAME !== sortBy,
+          })}
+          onClick={() => setSortBy(SortType.SORT_BY_NAME)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': SortType.SORT_BY_LENGTH !== sortBy,
+          })}
+          onClick={() => setSortBy(SortType.SORT_BY_LENGTH)}
+        >
```

Comment:
[UX] - As with the previous button, marking the active button would enhance user experience by indicating the current sorting method.